graphene/utils/tests/test_subclass_with_meta.py

```diff
@@ -0,0 +1,63 @@
+from ..subclass_with_meta import SubclassWithMeta
+
+
+def test_init_subclass_calls_super():
+    class DefinesInitSubclass:
+        subclass_init_count = 0
+
+        def __init_subclass__(cls, **kwargs):
+            DefinesInitSubclass.subclass_init_count += 1
+            super().__init_subclass__(**kwargs)
+
+    class InheritsAsFirst(SubclassWithMeta, DefinesInitSubclass):
+        pass
+
+    class InheritsAsLast(DefinesInitSubclass, SubclassWithMeta):
+        pass
+
+    assert DefinesInitSubclass.subclass_init_count == 2
+
+
+def test_kwargs_are_consumed():
+    class DefinesInitSubclass:
+        passed_kwargs = []
+
+        def __init_subclass__(cls, **kwargs):
+            if kwargs:
+                DefinesInitSubclass.passed_kwargs.append(kwargs)
+            super().__init_subclass__(**kwargs)
+
+    class InheritsAsFirst(SubclassWithMeta, DefinesInitSubclass, is_consumed="foo"):
```

Comment:
Is not clear to me what happens with the `is_consumed="foo"` meta variable.

I'm a bit concerned about the inconsistency between `InheritsAsFirst` and `InheritsAsLast`

Reply:
Well, that inconsistency is there even without this PR. I agree that it's a bit messy, maybe we could do:

```
try:
    super().__init_subclass__(**meta_options)
except TypeError:
    super().__init_subclass__()
```

Without the try-ecxept this will fail when `super()` is `object`, since it's `__init_subclass__` doesn't take any arguments.

Reply:
> ```
> try:
>     super().__init_subclass__(**meta_options)
> except TypeError:
>     super().__init_subclass__()
> ```

Okey, this is actually a bad way to do it, and can hide accidentally suppress unwanted errors. I think that it's fine that the `__init_subclass__` consumes all of the kwargs, since it really cannot know if the next class in the MRO is able to receive the kwargs.

Reply:
I need more time to think about this and come to a conclusion. Thanks for the analysis @ruohola

Reply:
Yeah, I truly think this is how the method should behave. The docs tell us https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__, that we are supposed to take the away needed kwargs and then continue the super chain. Since `SubclassWithMeta` is specifically made so that it takes use of all of the passed kwargs, we should take all of them away and just call the bare `super().__init_subclass__`.

Reply:
@syrusakbary So, have you made a decision?

Reply:
@syrusakbary So, what do you think?

Reply:
@syrusakbary @jkimbo So, what do you think?